### ISSUE TITLE
Durée de conservation des données conformément au RGPD : Intégration au sein de la DSI

### ISSUE THREAD

**Issue by Barbapapou** [correctexam/corrigeExamBack#59]
Afin de se conformer aux principes du RGPD, les données personnelles collectées par l'application doivent être régulièrement purgées. Il ne semble pas que cela soit actuellement le cas. Pour intégrer l'application au sein de la DSI, il serait nécessaire de mettre en place un tel système. Il conviendrait donc d'établir une durée maximale de conservation des données personnelles (nom, prénom, courriel, copie d'examen) dans la base de données de l'application.

**Comment by barais** [comment]
Oui clairement, je rajoute dans comme une feature request. Sais tu la durée . 

Genre purge tous les ans, c'est ok ?

**Comment by barais** [comment]
Cela pourrait être une bonne idée de permettre au passage un import et un export des données d'un exam/module afin si le prof le souhaite de pouvoir conserver offline ces données

**Comment by arnobl** [comment]
Je comprends la problématique RGPD, mais en tant qu'utilisateur je n'ai pas forcement envie que mes cours soient clean (sauf si la durée peut être très longue, comme 10 ans) : je peux vouloir garder les exams année sur année pour faire des stats, comparer (et que l'appli m'aide à comparer justement).
Aussi, on ne peut pas supprimer sans prévenir les utilisateurs.

**Comment by Barbapapou** [comment]
> 

Oui, une suppression chaque année serait suffisante. Je ne sais pas si le téléchargement des données en offline entre en conflit avec les règles RGPD.

**Comment by barais** [comment]
Ok, j'ai fait un premier jet pour faire un backup d'une UE afin qu'un prof qui souhaite conserver ses données puissent le faire. 
Je vais rajouter une capacité à détruire les données de plus de X mois (paramétrable au déploiement). Avec un mail X jours avant au prof pour le prévenir. 

Ce sera une première possibilité pour archiver pour les utilisateurs et détruire les données en ligne pour les étudiants. 

@Barbapapou Si tu as un contact pour voir si cela pourrait être ok avec la RGPD. (de détruire au bout d'un an).

**Comment by Barbapapou** [comment]
@barais 
Bonjour, désolé de la réponse tardive mais voici la réponse du DPO:
"Oui cette pratique peut entrer en contradiction avec le RGPD car des données personnelles d'étudiants "se baladent" alors sur divers supports de stockage rendant plus difficile leur encadrement ainsi que leur sécurité.
C'est aux scolarités "d'archiver" les résultats des étudiants. Il faudra sinon trouver une solution qui convient à tout le monde.
Serait il possible de faire un point avec un des enseignants concernés à la rentrée ? "
Je ne serai plus disponible à la rentrée, mais je peux vous mettre en contact.
